### ISSUE TITLE
[java] NonThreadSafeSingleton: False negative with ternary conditional operator

### ISSUE THREAD

**Issue by Chordrain** [pmd/pmd#6747]
**PMD Version:** 7.24.0
**Category:** [NonThreadSafeSingleton](https://pmd.github.io/pmd/pmd_rules_java_multithreading.html#nonthreadsafesingleton)
**Severity:** Warning (Multithreading)

## Summary

PMD's `NonThreadSafeSingleton` rule flags the unsafe lazy-initialization pattern `if (field == null) field = new T();` in a non-synchronized method (a classic check-then-act race).

When the same logic is rewritten as a ternary assignment — `field = field == null ? new T() : field;` — the rule **no longer fires** (false negative), even though the rewrite is semantically identical and exposes the very same data race under concurrent access.

### before — if-form (2 findings)

```java
public class LazyInitBefore {
    private static Object foo;
    private static Object bar;

    public static Object getFoo() {
        if (foo == null)
            foo = new Object();          // flagged
        return foo;
    }

    public static Object getBar() {
        if (bar == null)
            bar = new Object();          // flagged
        return bar;
    }
}
```

### after — ternary form (0 findings)

```java
public class LazyInitAfter {
    private static Object foo;
    private static Object bar;

    public static Object getFoo() {
        foo = foo == null ? new Object() : foo;   // FALSE NEGATIVE
        return foo;
    }

    public static Object getBar() {
        bar = bar == null ? new Object() : bar;   // FALSE NEGATIVE
        return bar;
    }
}
```

## Expected behavior

`NonThreadSafeSingleton` should report the same number of findings on BEFORE and AFTER. The ternary form `field = field == null ? new T() : field;` is semantically identical to `if (field == null) field = new T();` and has the same data-race window when invoked concurrently.

## Actual behavior

- Two if-form lazy inits → 2 findings (correct).
- Two ternary-form lazy inits → 0 findings.

The matcher only recognizes the `if`-statement shape of the lazy-init pattern and does not look through the equivalent conditional-assignment ternary.
